src/averageAges.js

```diff
@@ -37,8 +41,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+  const filteredPeople = withChildren
+    ? people.filter(x => people.some(({ mother }) => mother === x.name))
+    : people;
+
+  const womenAges = filteredPeople.filter(x => x.sex === 'f')
+    .map(({ born, died }) => died - born);
+  const sumOfAge = womenAges.reduce((prev, x) => prev + x);
```

Comment:
First - use correct indentations. If you chain several methods, each should start from new line;
Second - you can chain it with reduce too.

Same - in the other parts of task.
```suggestion
  const sumOfAges = filteredPeople
    .filter(x => x.sex === 'f')
    .map(({ born, died }) => died - born);
    .reduce((prev, x) => prev + x);
```

Reply:
Actually, I need womenAges.length to calculate the average in the end, that's why I didn't chain reduce. I'm not sure if I can get this info(number of people) out of .reduced sum.